src/docs/cookbook/testing/integration/introduction.md

```diff
@@ -134,187 +108,102 @@ class _MyHomePageState extends State<MyHomePage> {
 
 ### 2. Add the `flutter_driver` dependency
 
-Next, use the `flutter_driver` package to write integration tests.
-Add the `flutter_driver` dependency to the `dev_dependencies` section of
-the app's `pubspec.yaml` file.
-
-Also add the `test` dependency in order to use actual test functions and
-assertions.
+Next, use the `integration_test`, `flutter_driver`, and `flutter_test` packages
+to write integration tests. Add these dependencies to the `dev_dependencies`
+section of the app's `pubspec.yaml` file, specifying the Flutter SDK as the
+location of the package.
 
 ```yaml
 dev_dependencies:
+  integration_test:
+    sdk: flutter
   flutter_driver:
     sdk: flutter
-  test: any
+  flutter_test:
+    sdk: flutter
 ```
 
 ### 3. Create the test files
 
-Unlike unit and widget tests, integration test suites do not run in the same
-process as the app being tested. Therefore, create two files that
-reside in the same directory. By convention, the directory is named
-`test_driver`.
-
-  1. The first file contains an "instrumented" version of the app.
-     The instrumentation allows you to "drive" the app and record
-     performance profiles from a test suite. This file can have any
-     name that makes sense. For this example, create a file called
-    `test_driver/app.dart`.
-  2. The second file contains the test suite, which drives the app and
-     verifies that it works as expected. The test suite also records
-     performance profiles. The name of the test file must correspond
-     to the name of the file that contains the instrumented app,
-     with `_test` added at the end. Therefore,
-     create a second file called `test_driver/app_test.dart`.
-
-This creates the following directory structure:
+Create a new directory, `integration_test`, with an empty `app_test.dart` file:
 
 ```
 counter_app/
   lib/
     main.dart
-  test_driver/
-    app.dart
+  integration_test/
     app_test.dart
 ```
 
-### 4. Instrument the app
-
-Now, instrument the app. This involves two steps:
-
-  1. Enable the flutter driver extensions.
-  2. Run the app.
-
-Add the following code inside the
-`test_driver/app.dart` file.
-
-<?code-excerpt "test_driver/app.dart"?>
-```dart
-import 'package:flutter_driver/driver_extension.dart';
-import 'package:introduction/main.dart' as app;
-
-void main() {
-  // This line enables the extension.
-  enableFlutterDriverExtension();
-
-  // Call the `main()` function of the app, or call `runApp` with
-  // any widget you are interested in testing.
-  app.main();
-}
-```
-
-### 5. Write the tests
+### 4. Write the integration test
 
 Now that you have an instrumented app, you can write tests for it.
 This involves four steps:
 
-  1. Create [`SerializableFinders`][]
-     to locate specific widgets.
-  2. Connect to the app before our tests run in the `setUpAll()` function.
+  1. Initialize `IntegrationTestWidgetsFlutterBinding`, a singleton service that
+     executes tests on a physical device.
```

Comment:
Just want to point out that calling `IntegrationTestWidgetsFlutterBinding.ensureInitialized()`, and depending on `package:flutter_driver` in the `pubspec.yaml` isn't actually needed when running with `flutter test`. However, it is needed when running with `flutter drive` (required for web). Not sure if it will be cleaner to keep this, or move them to the section below on web though 🙂

Reply:
Nice! I vote we could move it to the web section

Reply:
Moved to the web section